adapters/repos/db/index.go

```diff
@@ -540,6 +566,12 @@ func (i *Index) IterateObjects(ctx context.Context, cb func(index *Index, shard
 // call ForEachLoadedShard instead.
 // Note: except Dropping and Shutting Down
 func (i *Index) ForEachShard(f func(name string, shard ShardLike) error) error {
+	// Check if the index is being dropped or shut down to avoid panics when the index is being deleted
+	if i.closingCtx != nil && i.closingCtx.Err() != nil {
```

Comment:
This is a race condition, as the index can start shutting down in between checking a state and using it, in a multithreaded environment.